osu.Game/Overlays/BeatmapListingOverlay.cs

```diff
@@ -240,6 +253,74 @@ private void load(TextureStore textures)
             }
         }
 
+        // using string literals as there's no proper processing for LocalizeStrings yet
+        public class SupporterRequiredDrawable : CompositeDrawable
+        {
+            private OsuSpriteText filtersText;
+
+            public SupporterRequiredDrawable()
+            {
+                RelativeSizeAxes = Axes.X;
+                Height = 225;
+                Alpha = 0;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(TextureStore textures)
+            {
+                AddInternal(new FillFlowContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    RelativeSizeAxes = Axes.Y,
+                    AutoSizeAxes = Axes.X,
+                    Direction = FillDirection.Horizontal,
+                    Children = new[]
+                    {
+                        new Sprite
+                        {
+                            Anchor = Anchor.Centre,
+                            Origin = Anchor.Centre,
+                            RelativeSizeAxes = Axes.Both,
+                            FillMode = FillMode.Fit,
+                            Texture = textures.Get(@"Online/supporter-required"),
+                        },
+                        createSupporterText(),
+                    }
+                });
+            }
+
+            public void UpdateText(List<LocalisableString> filters)
+            {
+                // use string literals for now
+                filtersText.Text = BeatmapsStrings.ListingSearchSupporterFilterQuoteDefault(string.Join(" and ", filters), "").ToString();
+            }
+
+            private Drawable createSupporterText()
+            {
+                LinkFlowContainer supporterRequiredText = new LinkFlowContainer
+                {
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    AutoSizeAxes = Axes.Both,
+                    Margin = new MarginPadding { Bottom = 10 },
+                };
+
+                filtersText = (OsuSpriteText)supporterRequiredText.AddText(
+                    "_",
+                    t =>
+                    {
+                        t.Font = OsuFont.GetFont(size: 16);
+                        t.Colour = Colour4.White;
+                    }
+                ).First();
```

Comment:
This doesn't have to be done this way. `AddText()` has an overload that accepts a `SpriteText`.

I'd also probably just skip this and recreate the entire text on every change. It doesn't really matter all that much for now, but it might later. Putting an entire string with spaces in a single `SpriteText`, which is in a `TextFlowContainer` makes it so that that text won't... flow. As long as there are no overly long variations of this string, that would begin to overflow the view, I guess it's passable for now.